contracts/Template.sol

```diff
@@ -1,106 +1,192 @@
+    * @param _financePeriod Initial duration for accounting periods, it can be set to zero in order to use the default of 30 days.
+    * @param _useAgentAsVault Boolean to tell whether to use an Agent app as a more advanced form of Vault app
+    */
+    function newInstance(
+        string memory _id,
+        address[] memory _holders,
+        uint256[] memory _stakes,
+        uint64[3] memory _votingSettings,
+        uint64 _financePeriod,
+        bool _useAgentAsVault
+    )
+        public
+    {
+        _validateId(_id);
+        _ensureTemplateSettings(_holders, _stakes, _votingSettings);
+
+        (Kernel dao, ACL acl) = _createDAO();
+        (Finance finance, Voting voting) = _setupBaseApps(dao, acl, _holders, _stakes, _votingSettings, _financePeriod, _useAgentAsVault);
+        _setupCustomApp(dao, acl, voting);
+        _transferCreatePaymentManagerFromTemplate(acl, finance, voting);
+        _transferRootPermissionsFromTemplateAndFinalizeDAO(dao, voting);
+        _registerID(_id, dao);
+    }
 
-        acl.createPermission(this, tokenManager, tokenManager.MINT_ROLE(), this);
-        tokenManager.mint(root, 1); // Give one token to root
+    function _setupBaseApps(
+        Kernel _dao,
+        ACL _acl,
+        address[] memory _holders,
+        uint256[] memory _stakes,
+        uint64[3] memory _votingSettings,
+        uint64 _financePeriod,
+        bool _useAgentAsVault
+    )
+        internal
+        returns (Finance, Voting)
+    {
+        MiniMeToken token = _popTokenCache(msg.sender);
+        Vault agentOrVault = _useAgentAsVault ? _installDefaultAgentApp(_dao) : _installVaultApp(_dao);
+        Finance finance = _installFinanceApp(_dao, agentOrVault, _financePeriod == 0 ? DEFAULT_FINANCE_PERIOD : _financePeriod);
+        TokenManager tokenManager = _installTokenManagerApp(_dao, token, TOKEN_TRANSFERABLE, TOKEN_MAX_PER_ACCOUNT);
+        Voting voting = _installVotingApp(_dao, token, _votingSettings);
+
+        _mintTokens(_acl, tokenManager, _holders, _stakes);
+        _setupBasePermissions(_acl, agentOrVault, voting, finance, tokenManager, _useAgentAsVault);
+
+        return (finance, voting);
+    }
 
-        acl.createPermission(ANY_ENTITY, voting, voting.CREATE_VOTES_ROLE(), root);
+    function _setupBasePermissions(
+        ACL _acl,
+        Vault _agentOrVault,
+        Voting _voting,
+        Finance _finance,
+        TokenManager _tokenManager,
+        bool _useAgentAsVault
+    )
+        internal
+    {
+        if (_useAgentAsVault) {
+            _createAgentPermissions(_acl, Agent(_agentOrVault), _voting, _voting);
+        }
+        _createVaultPermissions(_acl, _agentOrVault, _finance, _voting);
+        _createFinancePermissions(_acl, _finance, _voting, _voting);
+        _createFinanceCreatePaymentsPermission(_acl, _finance, _voting, address(this));
```

Comment:
We can use `_voting` as the manager here so we don't have to transfer the permission manager afterwards.

AFAIK this template doesn't need the Finance permission as it doesn't install Payroll.